tabulator.js

```diff
@@ -1260,15 +1260,32 @@
 
 		var top = typeof top == "undefined" ? self.options.addRowPos : (top === true || top === "top" ? "top" : "bottom");
 
+		// initial place to append the row is the table
+		var appendPlace = self.table;
+		// if group are used, look for the corresponding row of the group.
+		if(self.options.groupBy){
+			var groupVal = typeof(self.options.groupBy) == "function" ? self.options.groupBy(item) : item[self.options.groupBy];
+			var group = $(".tabulator-group[data-value='" + groupVal + "']", self.table);
+			//if group does not exist, build it
+			if(group.length == 0){
+				group = self._renderGroup(groupVal);
+				self.table.append(group);
+				self._renderGroupHeader(group);
+			}
+			//set the place to append the row to the corresponding group.
+			appendPlace = group;
+		}
 		//append to top or bottom of table based on preference
 		if(top == "top"){
-			self.activeData.unshift(item);
+			if (self.activeData != self.data)
```

Comment:
Please make sure all if statements use curly braces, it is better for code maintainability longer term.